webidls/Browser.webidl

```diff
@@ -0,0 +1,115 @@
+/**
+  * Browser is the equivalent of a top level Constellation::Frame.
+  * Browser is a top level browsing context.
+  * ~= Tab, ~= Webview
+  */
+
+[Constructor(DOMString contextName, StorageSession storageSession)]
+interface Browser {
+  // FIXME: how is this attached to a viewport? For swapping, it's important to be able to attach and detach
+
+  Promise<void> setHandler(BrowserHandler handler);
```

Comment:
Seems like questionable benefit at the cost of complexity. I would suggest to make a handler an argument of the constructor instead as it will accomplish as much but will keep things simpler and remove added asynchrony.